Aztec/Classes/Public/TextKit/TextStorage.swift

```diff
@@ -462,64 +462,24 @@ public class TextStorage: NSTextStorage {
         
         let originalLength = textStore.length
         textStore = NSMutableAttributedString(attributedString: output.attributedString)
-        
-        edited([.EditedAttributes, .EditedCharacters], range: NSRange(location: 0, length: originalLength), changeInLength: textStore.length - originalLength)
-    }
-
-    // MARK: - Image Handling
-    func loadImageForAttachment(attachment: TextAttachment, inRange range: NSRange) {
-
-        guard let imageProvider = imageProvider else {
-            fatalError("The image provider should've been set at this point.")
+        textStore.enumerateAttachmentsOfType(TextAttachment.self) { [weak self] (attachment, range, stop) in
+            attachment.imageProvider = self
         }
-
-        switch (attachment.kind) {
-        case .RemoteImage(let url):
-            attachment.image = imageProvider.storage(self, attachment: attachment, imageForURL: url, onSuccess: { [weak self] image in
-                self?.downloadSuccess(attachment, url: url, image: image)
-                }, onFailure: { [weak self] in
-                    self?.downloadFailure(attachment)
-            })
-        case .RemoteImageDownloaded(_, let image):
-            attachment.image = image
-
-        default:
-            attachment.image = imageProvider.storage(self, missingImageForAttachment: attachment)
-        }
-    }
-
-    private func downloadSuccess(attachment: TextAttachment, url: NSURL, image: UIImage) {
-
-        attachment.kind = .RemoteImageDownloaded(url: url, image: image)
-
-        attachment.image = image
-        invalidateLayoutForAttachment(attachment)
+        edited([.EditedAttributes, .EditedCharacters], range: NSRange(location: 0, length: originalLength), changeInLength: textStore.length - originalLength)
     }
+}
 
-    private func downloadFailure(attachment: TextAttachment) {
-        guard let imageProvider = imageProvider else {
-            fatalError("The image provider should've been set at this point.")
-        }
+extension TextStorage: TextAttachmentImageProvider {
 
-        attachment.image = imageProvider.storage(self, missingImageForAttachment: attachment)
-        invalidateLayoutForAttachment(attachment)
+    public func image(forURL url: NSURL, inAttachment attachment: TextAttachment,
```

Comment:
The `attachment` parameter is on the same line as the `url` parameter, while the other two are in new lines.